plugins/dbus_announce.c

```diff
@@ -103,9 +119,77 @@ static rpmRC send_ts_message(rpmPlugin plugin,
 
     dbcookie = rpmdbCookie(rpmtsGetRdb(ts));
     rpm_tid_t tid = rpmtsGetTid(ts);
+
+    if (!dbus_message_append_args(msg,
+				  DBUS_TYPE_STRING, &dbcookie,
+				  DBUS_TYPE_UINT32, &tid,
+				  DBUS_TYPE_INVALID))
+	goto err;
+
+    if (!dbus_connection_send(state->bus, msg, NULL))
+	goto err;
+
+    if (snprintf(detail_name, sizeof(detail_name), "%sDetails", name) >= sizeof(detail_name)) {
+        rpmlog(RPMLOG_WARNING,
+	       "dbus_announce plugin: Static buffer for details signal name is not large enough, needs at least %u bytes\n",
+	       (unsigned int)(strlen(name) + strlen("Details") + 1));
+        goto err;
+    }
+
+    dbus_message_unref(msg);
+
+    /* it's used in the log below, thus to have proper signal name in there */
+    name = detail_name;
+    msg = dbus_message_new_signal("/org/rpm/Transaction", /* object name */
+				  "org.rpm.Transaction",  /* interface name */
+				  name);                  /* signal name */
+    if (msg == NULL)
+	goto err;
+
+    nElems = rpmtsNElements(ts);
+    array = (char **) malloc((nElems + 1 ) * sizeof(char *));
```

Comment:
Already mentioned in another thread but please use `ARGV_t` here (and the accompanying functions such as `argvAdd()` and `argvFree()`).